Code/Mantid/Framework/WorkflowAlgorithms/src/ConvolutionFitSequential.cpp

```diff
@@ -0,0 +1,650 @@
+  plotPeaks->setAlwaysStoreInADS(true);
+  plotPeaks->setProperty("Input", plotPeakInput);
+  plotPeaks->setProperty("OutputWorkspace", outputWsName);
+  plotPeaks->setProperty("Function", function);
+  plotPeaks->setProperty("StartX", startX);
+  plotPeaks->setProperty("EndX", endX);
+  plotPeaks->setProperty("FitType", "Sequential");
+  plotPeaks->setProperty("CreateOutput", true);
+  plotPeaks->setProperty("OutputCompositeMembers", true);
+  plotPeaks->setProperty("ConvolveMembers", convolve);
+  plotPeaks->setProperty("MaxIterations", maxIter);
+  plotPeaks->setProperty("Minimizer", minimizer);
+  plotPeaks->setProperty("PassWSIndexToFunction", passIndex);
+  plotPeaks->executeAsChildAlg();
+  ITableWorkspace_sptr outputWs =
+      AnalysisDataService::Instance().retrieveWS<ITableWorkspace>(outputWsName);
+  // Delete workspaces
+  std::string deleteWorkspaces[] = {
+      (outputWsName + "_NormalisedCovarianceMatrices"),
+      (outputWsName + "_Parameters"), tempFitWsName};
+  auto deleter = createChildAlgorithm("DeleteWorkspace", 0.70, 0.73, true);
+  for (int i = 0; i < 3; i++) {
+    deleter->setProperty("WorkSpace", deleteWorkspaces[i]);
+    deleter->executeAsChildAlg();
+  }
+
+  // Construct output workspace
+  std::string resultWsName = outputWsName + "_Result";
+
+  // Define params for use in convertParametersToWorkSpace (Refactor to generic)
+  auto paramNames = std::vector<std::string>();
+  auto func = FunctionFactory::Instance().createFunction(funcName);
+  if (delta) {
+    paramNames.push_back("Height");
+  }
+  for (size_t i = 0; i < func->nParams(); i++) {
+    paramNames.push_back(func->parameterName(i));
+  }
+  if (funcName.compare("Lorentzian") == 0) {
+    // remove peak centre
+    size_t pos = find(paramNames.begin(), paramNames.end(), "PeakCentre") -
+                 paramNames.begin();
+    paramNames.erase(paramNames.begin() + pos);
+    paramNames.push_back("EISF");
+  }
+
+  // Run calcEISF if Delta
+  if (delta) {
+    auto columns = outputWs->getColumnNames();
+    calculateEISF(outputWs);
+  }
+
+  // Construct comma separated list for ProcessIndirectFirParameters
+  std::string paramNamesList = "";
+  const size_t maxNames = paramNames.size();
+  for (size_t i = 0; i < maxNames; i++) {
+    paramNamesList += paramNames.at(i);
+    if (i != (maxNames - 1)) {
+      paramNamesList += ",";
+    }
+  }
+
+  // Run ProcessIndirectFitParameters
+  auto pifp =
+      createChildAlgorithm("ProcessIndirectFitParameters", 0.73, 0.80, true);
+  pifp->setAlwaysStoreInADS(true);
+  pifp->setProperty("InputWorkspace", outputWs);
+  pifp->setProperty("ColumnX", "axis-1");
+  pifp->setProperty("ParameterNames", paramNamesList);
+  pifp->setProperty("OutputWorkspace", resultWsName);
+  pifp->executeAsChildAlg();
+
+  API::MatrixWorkspace_sptr resultWs =
+      AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(resultWsName);
+
+  // Set x units to be momentum transfer
+  auto axis = resultWs->getAxis(0);
```

Comment:
A true workflow algorithm should only run other algorithms. It looks like `ProcessIndirectFitParameters` can now take the X unit as an argument: https://github.com/mantidproject/mantid/blob/master/Code/Mantid/Framework/Algorithms/src/ProcessIndirectFitParameters.cpp#L71. Can we use that above instead of setting it manually?